backend/src/packages/chaiNNer_standard/image_filter/noise/fast_nlmeans.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import numpy as np
+import cv2
+
+from nodes.properties.inputs import ImageInput, SliderInput, NumberInput
+from nodes.properties.outputs import ImageOutput
+from nodes.groups import Condition, if_group
+from nodes.impl.image_utils import to_uint8
+from nodes.utils.utils import get_h_w_c
+
+from .. import noise_group
+
+
+@noise_group.register(
+    schema_id="chainner:image:fast_nlmeans",
+    name="Fast NL means",
+    description="Use the fast Non-Local Means algorithm to denoise an image.",
+    icon="CgEditNoise",
+    inputs=[
+        ImageInput("Image", channels=[1, 3, 4]),
+        SliderInput(
+            "Luminance strength",
+            minimum=0,
+            maximum=50,
+            default=3.0,
+            precision=1,
+            controls_step=0.1,
+            slider_step=0.1,
+        ),
+        if_group(Condition.type(0, "Image { channels: 3 | 4 }"))(
+            SliderInput(
+                "Color strength",
+                minimum=0.,
+                maximum=50.,
+                default=3.,
+                precision=1,
+                controls_step=0.1,
+                slider_step=0.1,
+            )
+        ),
+        NumberInput("Patch radius",
+            minimum=1,
+            default=3,
+            maximum=30,
```

Comment:
Why is the minimum 1? We use 2r+1, so shouldn't the radius' minimum be 0, so we can get a size of 1?

Why is maximum 30? It was 49 before, which would translate to a 24. Is the maximum just an arbitrary number? I'm not against having a defined maximum, not allowing inputs that don't make sense is a good thing, I'm just curious why this number.

Same for search radius.

Reply:
1. error from my side. However, not sure it makes sense as I have never saw this values in any paper.
https://www.apeer.com/app/modules/Non-local-means-denoising/da13051a-db17-4b97-a260-8faa9ac341b7
let me know.

2. Maximum is an arbitrary nb.Never seen such high values in papers. (window size  = 2*radius+1)
https://jivp-eurasipjournals.springeropen.com/articles/10.1186/s13640-018-0346-y
I do not see why patch size would be > search size, but did not implement a condition as I am not an expert.
 I am open to any suggestions.

Reply:
> However, not sure it makes sense as I have never saw this values in any paper.

Then let's keep it at 1.

> Never seen such high values in papers.

Good enough for me. I was just curious.